### ISSUE TITLE
Enhancement: Synthesise notification badges for most applications (using messageTray)

### ISSUE THREAD

**Issue by xalt7x** [micheleg/dash-to-dock#1861]
Since most major free and commercial Desktop Environments switched to "Icons only" style for showing application windows on docks/panels , notification badges had become a very welcome feature, especially for the instant messages (chat) applications.
Unity DE had an API for badges. KDE Plasma and some GNOME extensions ([including](https://github.com/micheleg/dash-to-dock/pull/590)) still support it. But the problem is that application support is rather limited. Among chat applications I know only Telegram and Ferdium). And I highly doubt that most other Electron or even Chromium PWA based applications (like official versions of Google Chat or Microsoft Teams) will ever officially adopt Unity API. Not to mention most non-Ubuntu-based distros where it's barely supported.

---

I've seen that other GNOME extensions implemented Unity-independent badges/counters (see references).
Basically they add badges /counters for applications which send notifications to the GNOME Message Tray.

References:
* [RocketBar](https://github.com/linux-is-awesome/gnome_extension_rocketbar) (works very well and has optional support for Unity API)
* [Notification Counter](https://github.com/vkrizan/NotificationCounter)
* [new-mail-indicator](https://github.com/fthx/new-mail-indicator)
* [Unofficial modification for "Dash to Panel"](https://github.com/home-sweet-gnome/dash-to-panel/issues/1695#issuecomment-1160077515)

**Comment by vanvugt** [comment]
It sounds like the problem you're reporting is that apps don't support badges. Do you mean to report that the dock does not support badges?

**Comment by Alt37** [comment]
Yeah, such issues pop from time to time in various places.
As it turned out the problem is even bigger with "sandboxing".
At least some Snap* & Flatpak** packages currently can't use UnityAPI even if application supports it

* https://github.com/signalapp/Signal-Desktop/issues/4982
** https://github.com/flathub/com.discordapp.Discord/issues/228

So at one hand we could use this kind of hacks which upstream DE wouldn't accept for obvious reasons,
at the other - not a sign that this problem would be solved in a distant future.
Honestly, as a user I would prefer using a "hack"  :)
P.S. @vanvugt  , thanks for keeping this thread alive!

**Comment by paroj** [comment]
>  At least some Snap* & Flatpak** packages currently can't use UnityAPI even if application supports it

speaking of snap, this is an application packaging bug. I have used the UnityAPI in my snaps and it worked since snaps first appeared
